keps/sig-api-machinery/storage-migration.md

```diff
@@ -141,29 +149,57 @@ Read the [workflow section][] to understand how the API is used.
 ```golang
 type StorageVersionMigration struct {
   metav1.TypeMeta
-  // metadata.name should be "<resource>.<group>" of the resource being
-  // migrated.
-  // Because resouces can be served at different groups, in the future version,
-  // metadata.name can be hash(etcd instance, etcd path prefix).
+  // At the alpha stage, metadata.name must be "<resource>.<group>" of the
+  // resource being migrated.
+  // We require this name format to avoid duplicate `migration` CRs for the same
+  // resource type. Otherwise we will need to build complicated system to make
+  // sure the latest `migration` CR is honored. 
```

Comment:
You already execute only one migration at a time; why would you need to build anything else?

Reply:
Why is that problematic? Do you expect clients to request that? Folks have repeatedly said we don't care about extra migrations.

Also if the client wants to do that, they can always just repeatedly delete and re-add the object.

We don't need to protect them from themselves to that degree.